core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultLockManager.java

```diff
@@ -389,5 +432,22 @@ CompositeLockPromise scheduleLockTimeoutTask(ScheduledExecutorService executorSe
          }
          return this;
       }
+
+      private <T> T checkState(LockState state, Supplier<T> acquired, Function<Throwable, T> exception) {
```

Comment:
I find the name a bit misleading. I would expect it to always return immediately, but if `state == WAITING` it will block.

Reply:
it can't be. the method is invoked when the notifier is completed. the state can be acquired, timed_out or deadlock (I guess the deadlock can be removed)